doc/source/serve/doc_code/vllm_example.py

```diff
@@ -0,0 +1,88 @@
+import json
+import random
+from typing import AsyncGenerator
+from starlette.requests import Request
+from starlette.responses import StreamingResponse, Response
+
+from transformers import AutoTokenizer
+
+from ray import serve
+
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from vllm.sampling_params import SamplingParams
+from vllm.utils import random_uuid
+
+
+@serve.deployment(ray_actor_options={"num_gpus": 1})
+class VLLMPredictDeployment:
+    def __init__(self, **kwargs):
+        args = AsyncEngineArgs(**kwargs)
+        self.engine = AsyncLLMEngine.from_engine_args(args)
+        
+    async def __call__(self, request: Request) -> Response:
+        """Generate completion for the request.
+
+        The request should be a JSON object with the following fields:
+        - prompt: the prompt to use for the generation.
+        - stream: whether to stream the results or not.
+        - other fields: the sampling parameters (See `SamplingParams` for details).
+        """
+        request_dict = await request.json()
+        prompt = request_dict.pop("prompt")
+        stream = request_dict.pop("stream", False)
+        sampling_params = SamplingParams(**request_dict)
+        request_id = random_uuid()
```

Comment:
@sihanwang41 we should make `request_id` publicly-accessible for use cases like this